DisCatSharp.Interactivity/Extensions/InteractionExtensions.cs

```diff
@@ -51,5 +53,72 @@ public static class InteractionExtensions
 		/// <param name="token">A custom cancellation token that can be cancelled at any point.</param>
 		public static Task SendPaginatedResponseAsync(this DiscordInteraction interaction, bool ephemeral, DiscordUser user, IEnumerable<Page> pages, PaginationButtons buttons = null, PaginationBehaviour? behaviour = default, ButtonPaginationBehavior? deletion = default, CancellationToken token = default)
 			=> MessageExtensions.GetInteractivity(interaction.Message).SendPaginatedResponseAsync(interaction, ephemeral, user, pages, buttons, behaviour, deletion, token);
+
+		/// <summary>
+		/// Sends multiple modals to the user with a prompt to open the next one.
+		/// </summary>
+		/// <param name="interaction">The interaction to create a response to.</param>
+		/// <param name="builder">The paginated modal builder.</param>
+		/// <param name="timeOutOverride">A custom timeout. (Default: 15 minutes)</param>
+		/// <returns>A read-only dictionary with the customid of the components as the key.</returns>
+		/// <exception cref="ArgumentException">Is thrown when no modals are defined.</exception>
+		/// <exception cref="InvalidOperationException">Is thrown when interactivity is not enabled for the client/shard.</exception>
+		public static async Task<PaginatedModalResponse> CreatePaginatedModalResponseAsync(this DiscordInteraction interaction, DiscordInteractionPaginatedModalBuilder builder, TimeSpan? timeOutOverride = null)
+		{
+			if (builder.Modals is null || builder.Modals.Count is 0)
+				throw new ArgumentException("You have to set at least one page");
+
+			var client = (DiscordClient)interaction.Discord;
+			var interactivity = client.GetInteractivity() ?? throw new InvalidOperationException($"Interactivity is not enabled for this {(client.IsShard ? "shard" : "client")}.");
+
+			timeOutOverride = timeOutOverride is null ? TimeSpan.FromMinutes(15) : timeOutOverride;
+
+			Dictionary<string, string> caughtResponses = new();
+
+			DiscordInteraction previousInteraction = interaction;
+
+			foreach (var b in builder.Modals)
+			{
+				var modal = b.Modal.WithCustomId(Guid.NewGuid().ToString());
+
+				DiscordMessage originalResponse = null;
+
+				if (previousInteraction.Type is InteractionType.Ping or InteractionType.ModalSubmit)
+				{
+					await previousInteraction.CreateResponseAsync(InteractionResponseType.ChannelMessageWithSource, b.OpenMessage.AddComponents(b.OpenButton));
+					originalResponse = await previousInteraction.GetOriginalResponseAsync();
+					var modalOpen = await interactivity.WaitForButtonAsync(originalResponse, new List<DiscordButtonComponent> { b.OpenButton }, timeOutOverride);
+
+					if (modalOpen.TimedOut)
+					{
+						_ = previousInteraction.EditOriginalResponseAsync(new DiscordWebhookBuilder().WithContent(b.OpenMessage.Content).AddComponents(b.OpenButton.Disable()));
```

Comment:
What is the purpose on assigning to `_` here?

Reply:
To simply throw away the response, it's not necessary to await whether it was done or not.